src/mainwindow.cpp

```diff
@@ -729,15 +729,27 @@ void MainWindow::buildTranslateSelectionState(QState *state) const
     auto *showWindowState = new QState(state);
     auto *translationState = new QState(state);
     auto *finalState = new QFinalState(state);
-    state->setInitialState(setSelectionAsSourceState);
+
+    // On Wayland, the clipboard/selection content can only be obtained if the window is active
+    // so we need to show the window first
+    if (QGuiApplication::platformName() == QStringLiteral("wayland")) {
```

Comment:
Maybe always show window first independent of Wayland presence? I just tested it on X11, this logic works on X11.